internal/integration_test/filecache/filecache_test.go

```diff
@@ -0,0 +1,66 @@
+package filecache
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/tetratelabs/wazero/experimental"
+	"github.com/tetratelabs/wazero/internal/engine/compiler"
+	"github.com/tetratelabs/wazero/internal/integration_test/spectest"
+	v1 "github.com/tetratelabs/wazero/internal/integration_test/spectest/v1"
+	"github.com/tetratelabs/wazero/internal/platform"
+	"github.com/tetratelabs/wazero/internal/testing/require"
+)
+
+func TestSpecTestCompilerCache(t *testing.T) {
+	if !platform.CompilerSupported() {
+		return
+	}
+
+	const cachePathKey = "FILE_CACHE_DIR"
+	cacheDir := os.Getenv(cachePathKey)
+	if len(cacheDir) == 0 {
+		// This case, this is the parent of the test.
+		cacheDir = t.TempDir()
+
+		// Before running test, no file should exist in the directory.
+		files, err := os.ReadDir(cacheDir)
+		require.NoError(t, err)
+		require.True(t, len(files) == 0)
+
+		// Get the executable path of this test.
+		testExecutable, err := os.Executable()
```

Comment:
so this is the key - with this, we can os.Exec this test itself recursively regardless of this is executed via `go test ./...` or as the stand alone binary as we do with scratch containers.

Reply:
cool!